lib/filestorage.js

```diff
@@ -9,92 +9,141 @@ const CHECKSUM = 'CRC32';
   //         before FileStorage is used
-  //     minCompressSize <number> minimal file size
+  //     minCompressSize - <number>, minimal file size
   //         to be compressed, default = 1024
   constructor(options) {
     this.dir = path.resolve(options.dir);
     this.minCompressSize = options.minCompressSize || MIN_COMPRESS_SIZE;
   }
 
   // Write file to storage
-  //   id <Object> common.Uint64, id of file
-  //   data <string> | <Buffer> | <Uint8Array> data to be written
-  //   opts <Object>
-  //     checksum <string> checksum type
-  //     dedupHash <string> second checksum type
-  //   cb <Function> callback
-  //     err <Error>
-  //     checksum <string> data checksum
-  //     dedupHash <string> second data checksum
-  //     originalSize <number> data size
-  // Throws: <TypeError> if `opts.checksum` or `opts.dedupHash` is incorrect
+  //   id - <common.Uint64>, id of file
+  //   data - <string> | <Buffer> | <Uint8Array>, data to be written
+  //   opts - <Object>
+  //     checksum - <string>, checksum type
+  //     dedupHash - <string>, second checksum type
+  //   cb - <Function>, callback
+  //     err - <Error>
+  //     stats - <Object>
+  //       checksum - <string>, data checksum
+  //       dedupHash - <string>, second data checksum
+  //       size - <number>, data size
+  // Throws: <TypeError>, if `opts.checksum` or `opts.dedupHash` is incorrect
   write(id, data, opts, cb) {
     const checksum = utils.computeHash(data, opts.checksum || CHECKSUM);
     const dedupHash = utils.computeHash(data, opts.dedupHash || DEDUP_HASH);
-    const originalSize = Buffer.byteLength(data);
+    const size = Buffer.byteLength(data);
 
-    const file = utils.getFilepath(this.dir, common.idToPath(id));
+    const file = this[getFilepath](id);
     utils.mkdirp(path.dirname(file), err => {
       if (err) {
         cb(err);
         return;
       }
       fs.writeFile(file, data, err => {
         if (err) cb(err);
-        else cb(null, checksum, dedupHash, originalSize);
+        else cb(null, { checksum, dedupHash, size });
       });
     });
   }
 
+  // Update file in the storage
+  //   id - <common.Uint64>, id of file
+  //   data - <string> | <Buffer> | <Uint8Array>, data to be written
+  //   opts - <Object>
+  //     checksum - <string>, checksum type
+  //     dedupHash - <string>, second checksum type
+  //   cb - <Function>, callback
+  //     err - <Error>
+  //     stats - <Object>
+  //       checksum - <string>, data checksum
+  //       dedupHash - <string>, second data checksum
+  //       size - <number>, data size
+  //       originalSize - <number>, size of original file
+  // Throws: <TypeError>, if `opts.checksum` or `opts.dedupHash` is incorrect
+  update(id, data, opts, cb) {
+    const file = this[getFilepath](id);
+    fs.stat(file, (err, fstats) => {
+      if (err) {
+        cb(err);
+        return;
+      }
+      this.write(id, data, opts, (err, stats) => {
```

Comment:
I don't think we need to go though `utils.mkdirp` again and can `fs.writeFile(` right away.
As this assumes (and checks) that file exists.